pages/resources/[page].js

```diff
@@ -194,11 +194,8 @@ function Resources() {
                 handleSubmit(values, actions);
                 actions.setSubmitting(true);
               }}
-              onReset={(values, actions) => {
-                handleReset(values, actions);
-              }}
             >
-              {({ isSubmitting }) => (
+              {({ isSubmitting, resetForm }) => (
                 <Form role="search">
```

Comment:
```suggestion
                <Form role="search" onReset={resetForm}>
```

to be clear, you tried this and it didnt work?

Reply:
yes, it led to Maximum callstack exceeded error. 
* I would click "Search" - component would re-render, leading to onReset being triggered (for no reason?) and onReset would reset the form, which would lead some re-render and so on...
* the same when you clicked "Reset" - it would lead to endless loop with `onReset` being called on and on

Reply:
I one thought I have is that when you do `formikBag.resetForm()` the side effect would `onReset` being triggered, which I think was not the initial intention but could be a side-effect that's causing this

Reply:
@kylemh I guess I'm right here, take a look at the implementation of [`resetForm`](https://github.com/formium/formik/blob/master/packages/formik/src/Formik.tsx#L378) there is a condition `if(props.onReset)` inside which calls `onReset` callback.
So if we have a `onReset` calling `resetForm` it will lead to endless loop.

Reply:
AH! https://github.com/Tomekmularczyk/front-end/blob/chore/update-formik/components/Form/Form.js

Looks like we may need to edit some stuff here then

Reply:
I think it's fine, the props are correct and `onReset` callback was only used in this file alone

Reply:
I think it's because it's the only form with a reset button.

Reply:
What if we have _no_ callback for reset handler (including the button) and just let it by `type="reset"`.

Reply:
Just checked, nothing will happen. If we want to keep `onReset` we need to find a way not to call `resetForm` in it

Reply:
In the original issue, I think I referenced getting rid of <Formik> usages, didn't I? I feel like if we're leveraging the `useFormik` hook we may entirely ditch the `Form` component, no?

Reply:
true :D just noticed that, though I think updating the dependency and refactoring all forms could be separate issues 😅  seems like a lot of work

Reply:
What is more `onReset` is being called even when the form is reinitialized so

* select something in the form 
* press submit
* URL is updated
* component re-renders and initialValues change
* Formik reinitializes because of new initialValues and `enableReinitialize` prop
* onReset is being called because of that reinitialization
* `router.push(pathname, '/resources/1', { shallow: true });` is called and URL resets
* because URL is different again, initialValues change again and form is reinitialized to initial - empty - state

Seems like `onReset` has a very different purpose now than it had in v1. IMO, I would remove it.

@kylemh let me know if there's anything I could/should do with this because I don't think there's much more I can do :)

Reply:
it sounds like future PRs to refactor the forms would be best

Reply:
once the resources API is back up and running I'll merge this

if you wanna do any of what we discussed above, feel free to branch off of this

Reply:
thank you a lot :) for now, I think I made my share of work in this repo, I will leave refactoring to hooks for other Hacktoberfest participants :)